File: Easy/88. Merge Sorted Array/88. Merge Sorted Array.swift
class Solution {
    func merge(_ nums1: inout [Int], _ m: Int, _ nums2: [Int], _ n: Int) {
        var pos = m+n-1
        var m = m-1
        var n = n-1

        while n >= 0 {
            if m < 0 || nums1[m] < nums2[n] {
                nums1[pos] = nums2[n]
                n -= 1
            } else {
                nums1[pos] = nums1[m]
                m -= 1
            }
            pos -= 1
        }
    }
}

File: Easy/88. Merge Sorted Array/88. Merge Sorted Array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int r = nums1.size() - 1;
        int p = m - 1;
        int q = n - 1;
        while(p >= 0 && q >= 0) {
            if ( nums1[p] > nums2[q]) {
                nums1[r--] = nums1[p--];
            } else {
                nums1[r--] = nums2[q--];
            }
        }
        while (q >= 0){
            nums1[r--] = nums2[q--];
        }
    }
};